tailcfg/tailcfg.go

```diff
@@ -161,11 +161,9 @@ type Node struct {
 	StableID StableNodeID
 	Name     string // DNS
 
-	// DisplayName is the title to show for the node in client
-	// UIs. This field is assigned by default in controlclient,
-	// but can be overriden by providing this field non-empty
-	// in a MapResponse.
-	DisplayName string `json:",omitempty"`
+	// displayName contains name info for this node, cached for
+	// easy access in node.DisplayName and node.DisplayNames.
+	displayName displayName
```

Comment:
Is there any harm in making this non-hidden? Asking because _I think_ that's the only way we'll be able to use the displayName logic directly in iOS and macOS. Unless there's some way for us to add function calls to IPN.Swift (https://github.com/tailscale/corp/pull/1211/files#diff-b7188f8497f2fb0511eecddb0dfc8b2b2f3758e57025a368a7219b465995bc0cR42)


We always initialize displayName from controlclient so control would not be able to override it.

Reply:
Then we need to also export the DisplayName type and make it a pointer and not a valid (otherwise omitempty won't work).

But at that point I'd ditch the type and just have N string fields, with docs on each saying they're not from control.

I'd probably name they call with some obvious ugly prefix to make that really obvious, like `ComputedName string` or `ComputedHostnameIfDifferent string`.

Reply:
Before I do that, I just want to make sure - there's no way to access our `DisplayName` or `DisplayNames` functions directly from swift right?

Reply:
Not currently. Also they're in a different process, adding to the fun.

Reply:
Hahaha alright cool, I guess I'll go with the `ComputedName string`, `ComputedHostnameIfDifferent string` etc route to avoid repeating the logic in swift.